### DISCUSSION TITLE
Finally, V5 is here, 3x faster, supporting 6000+ languages!

### DISCUSSION THREAD

**Discussion by snakers4** [snakers4/silero-vad#471] [General]
# **Finally, V5 is here, 3x faster, supporting 6000+ languages!**

![image](https://github.com/snakers4/silero-vad/assets/12515440/5f581592-3035-411e-9803-44a2fda0af8e)

## **Performance and Model Size**

- 3x [faster](https://github.com/snakers4/silero-vad/wiki/Performance-Metrics) inference for TorchScript, 10% faster inference for ONNX;
- Now TorchScript is as fast as ONNX;
- Model size is 2x larger, 2MB vs. 1MB;

## **Quality**

- The VAD supports more than 6,000 languages now;
- Significanly more robust on noisy data;
- Overall 5-7% quality increase on clean data;
- Quality difference for 8 kHz and 16 kHz is negligible now;
- Quality difference for different window sizes is negligible => window size was deprecated;
- Added [benchmarks ](https://github.com/snakers4/silero-vad/wiki/Quality-Metrics) on 9 unique datasets (2 private) and one holistic multi-domain dataset;

## **Changes and deprecations**

- ONNX opset 16;
- `window_size_samples` is deprecated - now the VAD only works with fixed size window;
- VAD now works with 8 kHz and 16 kHz sample rates, only with fixed 256 and 512 sample windows respectively;
- Slightly changed internal logic, now some context (part of previous chunk) is passed along with the current chunk;
- Sample rates that are a multiple of 16 kHz are still supported;

<hr /><em>This discussion was created from the release <a href='https://github.com/snakers4/silero-vad/releases/tag/v5.0'>Finally, V5 is here, 3x faster, supporting 6000+ languages!</a>.</em>

**Comment by dgoryeo** [comment]
🎉 Congratulations on the release of Silero VAD Version 5! 🎉

Finally, V5 is here! This is a much-anticipated milestone and a testament to the hard work and dedication of you, @snakers4, and the team. Keep up the fantastic work!

**Comment by hoonlight** [comment]
Really awesome!

**Comment by RiccardoRomagnoli** [comment]
Awesome work!!

**Comment by smallbraineng** [comment]
I am noticing degraded performance on this model–am I passing stateN correctly (is it equivalent to how "h" and "c" were passed earlier?)

specifically, how to implement this? "now some context (part of previous chunk) is passed along with the current chunk"

**Reply by snakers4** [reply]
If you are using the ONNX version, then see changes to this function - https://github.com/snakers4/silero-vad/blob/fdbb0a3a81e0f9d95561d6b388d67dce5d9e3f1b/utils_vad.py#L9-L105

For TorchScript version - the interface is hidden inside of the model and the invocation should remain the same.

**Comment by Hiniu6996** [comment]
> # **Finally, V5 is here, 3x faster, supporting 6000+ languages!**
> 
> ![image](https://github.com/snakers4/silero-vad/assets/12515440/5f581592-3035-411e-9803-44a2fda0af8e)
> 
> ## **Performance and Model Size**
> 
> - 3x [faster](https://github.com/snakers4/silero-vad/wiki/Performance-Metrics) inference for TorchScript, 10% faster inference for ONNX;
> - Now TorchScript is as fast as ONNX;
> - Model size is 2x larger, 2MB vs. 1MB;
> 
> ## **Quality**
> 
> - The VAD supports more than 6,000 languages now;
> - Significanly more robust on noisy data;
> - Overall 5-7% quality increase on clean data;
> - Quality difference for 8 kHz and 16 kHz is negligible now;
> - Quality difference for different window sizes is negligible => window size was deprecated;
> - Added [benchmarks ](https://github.com/snakers4/silero-vad/wiki/Quality-Metrics) on 9 unique datasets (2 private) and one holistic multi-domain dataset;
> 
> ## **Changes and deprecations**
> 
> - ONNX opset 16;
> - `window_size_samples` is deprecated - now the VAD only works with fixed size window;
> - VAD now works with 8 kHz and 16 kHz sample rates, only with fixed 256 and 512 sample windows respectively;
> - Slightly changed internal logic, now some context (part of previous chunk) is passed along with the current chunk;
> - Sample rates that are a multiple of 16 kHz are still supported;
> 
> <hr /><em>This discussion was created from the release <a href='https://github.com/snakers4/silero-vad/releases/tag/v5.0'>Finally, V5 is here, 3x faster, supporting 6000+ languages!</a>.</em>

**Comment by jnhck** [comment]
Thank you for all your work. This is an awesome project!

**Comment by ksyntiant** [comment]
Hi I just started downloading data mentioned in the datasets section. I downloaded 3 out of 5 datasets and am already at around 1.8TB of data. I am just wondering how much TB of data was v5 model trained on. Is it possible to get more information on your training data other than provided in the datasets page. Thanks :)

**Comment by kormang** [comment]
@snakers4 Great job. Just discovered this and it's awesome!

However, it would have been better if you used 320 or 640 samples window, instead of 512. Because that corresponds to 20ms (or 40ms) of 16kHz audio, which is standard packet size for real-time communication.

Maybe something to think about for V6.

**Comment by Harshits-meesho** [comment]
Can someone tell me how segment size is 31.25 ms?

**Reply by snakers4** [reply]
512 frames with an SR of 16k

**Reply by Harshits-meesho** [reply]
@snakers4 , Thanks for the prompt response!
That is 32 ms, right? From where 31.25 number is coming? any formula?
(I am new to speech data. Sorry, If my question seems to be naive 😅

**Reply by thet0ast3r** [reply]
16000/(1024*500)
i assume framesize 1024
